modules/juce_audio_devices/native/juce_ios_Audio.cpp

```diff
@@ -573,6 +572,52 @@ struct iOSAudioIODevice::Pimpl      : public AudioPlayHead,
 
         return session.inputGain == gain;
     }
+    
+    void selectBackMic ()
+    {
+        auto session = [AVAudioSession sharedInstance];
+
+        NSArray *portDescriptions = session.availableInputs;
+        AVAudioSessionPortDescription* builtInMicPort = nil;
+
+        for (AVAudioSessionPortDescription* port in portDescriptions) {
+            if ([port.portType isEqualToString:AVAudioSessionPortBuiltInMic]) {
+                builtInMicPort = port;
+                break;
+            }
+        }
+
+        if (builtInMicPort) {
+            for (AVAudioSessionDataSourceDescription* source in builtInMicPort.dataSources) {
+                if ([source.orientation isEqual:AVAudioSessionOrientationBack]) {
+                    break;
+                }
+            }
+        }

```

Comment:
I'm slightly confused here. Where are we actually selecting the mic choice? It kind of seems like we're just checking state, but not setting anything. Am I misunderstanding here?

Reply:
Good catch. The mic position was changed because I'm using `AVAudioSessionModeVideoRecording`, not because of this function

Reply:
So then do we still need this function?

Reply:
Yes, my iPhone 6s selects the top front microphone instead of the back one so we need to force to use the back mic across all models